plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/WarPublisher.java

```diff
@@ -71,12 +70,7 @@ private static void publish(IProject project, IPath destination, boolean explode
     progress.setTaskName(Messages.getString("task.name.publish.war"));
 
     PublishHelper publishHelper = new PublishHelper(null);
-    J2EEFlexProjDeployable deployable =
-        new J2EEFlexProjDeployable(project, ComponentCore.createComponent(project));
-
-    List<IModuleResource> resourcesToDeploy = Lists.newArrayList(deployable.members());
-    resourcesToDeploy.addAll(computeWebFragmentJarResources(deployable, progress));
-    IModuleResource[] resources = resourcesToDeploy.toArray(new IModuleResource[0]);
+    IModuleResource[] resources = flattenResources(project, progress);
 
     if (exploded) {
       publishHelper.publishFull(resources, destination, progress.newChild(100));
```

Comment:
Good catch. There was a different. For example, `publishSmart()` was deleting existing files not known to itself, so if we put a web frag JAR into `WEB-INF/lib` first and later it tries to publish the main lib JARs into `WEB-INF` (i.e., if the web frag JAR comes first in the `IModuleResource[]` list), `publishSmart()` clears the web frag. I figured out we always create a new directory each time, so I guess there's no reason to use `publishSmart()`.